### DISCUSSION TITLE
특정 상황에서 소켓이 끊어집니다.

### DISCUSSION THREAD

**Discussion by h2saj5xz** [storycraft/node-kakao#271] [Q&A]
현재 재현가능한 상황은
맥 이외의 OS에서 구동한 상태로
채널 데이터가 큰 계정에 클라이언트 로그인했을때이며
read ECONNRESET at TCP.onStreamRead 에러가 발생합니다.

채널 데이터가 작은 계정은 항상 잘 작동합니다.
문제가 발생하는 계정의 경우 윈도 재부팅후 첫번째는 로그인이 되지만
두번째는 100% 문제가 발생합니다.

실제 PC 윈도 파워쉘에서 v14 노드로 실행했고
SM-T976N 안드로이드 10.0 카카오 9.2.1 셋팅입니다.

인터넷 환경이 다른 두곳에서 테스트했으나 같은 증상이 있었고
부트캠프 윈도랑 리눅스(시놀로지)에서도 마찬가지였지만,
맥에서 테스트 했을때는 정상 작동했습니다.

아래 코드를 실행할때
loginRes = await talkClient.login(authToken); 

아래의 에러가 발생합니다.
Error: read ECONNRESET
    at TCP.onStreamRead (internal/stream_base_commons.js:209:20)
(node:15168) UnhandledPromiseRejectionWarning: Error: read ECONNRESET
    at TCP.onStreamRead (internal/stream_base_commons.js:209:20)
(node:15168) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:15168) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

실제 윈도 카카오 클라이언트에서는 문제가 없으니
뭔가 설정이 있을 것 같은데, 어떻게 해야할지 모르겠네요.

**Comment by h2saj5xz** [comment]
OS 베이스에서 소켓 버퍼를 로그인시 받아오는 채널 데이터보다 크게 설정한뒤 작동하는 것을 확인했습니다.

**Comment by generalxd** [comment]
etc/sysctl.conf 파일 수정했더니 해결됐습니다.
